### ISSUE TITLE
Q: Is it possible to configure API Gateway stage variables?

### ISSUE THREAD

**Issue by ottokruse** [aws/chalice#728]
Is it possible to configure API Gateway stage variables? E.g. set them up in the config.json and have them set up in API Gateway upon "chalice deploy"?

Context: we use a lambda authorizer that use stage variables.

**Comment by kadrach** [comment]
I don't think this is possible directly, Chalice uses Lambda environment variables.

If you use the CloudFormation deployer you can modify the generated SAM template before deployment. I right-merge a custom config file on top of it for stage variables. It's a hack, but it works :)

**Comment by ottokruse** [comment]
Aw that is too bad! Thanks for the tip though.

However is this something that could/should someday (soon) be possible in Chalice. Or is it "by design" that everything works with lambda environment variables instead of api gateway stage variables?

I would like to re-use my Authorizer lambda function in multiple API Gateway API's. I don't see another way of passing configuration to the authorizer other than through stage variables

**Comment by stealthycoin** [comment]
I can see how this would be useful. Adding this as a feature request.

**Comment by wutachih** [comment]
So if we want to use custom domain, we will have to do it from console or cli.
